sdk/keyvault/azure-security-keyvault-secrets/swagger/autorest.md

```diff
@@ -39,13 +39,25 @@ namespace: com.azure.security.keyvault.secrets
 models-subpackage: implementation.models
 custom-types-subpackage: models
 enable-sync-stack: true
-generate-client-interfaces: false
 generate-client-as-impl: true
-service-interface-as-public: true
 license-header: MICROSOFT_MIT_SMALL
 disable-client-builder: true
-add-context-parameter: true
-context-client-method-parameter: true
-generic-response-type: true
-stream-style-serialization: true
+```
+
+### Rename SecretSetParameters.contentType and SecretUpdateParameters.contentType to secretContentType
+
+This solves an issue with generators after 4.1.29 (uncertain of which version as the update went from 4.1.29 to 4.1.42)
+where in the generated APIs using these types as parameters the previous constant for the request content type of
+`String contentType = "application/json"` was removed and replaced with the `contentType` value for `SecretSetParameters`
+or `SecretUpdateParameters`. Obtusely, this change causes the interface method to no longer add `@HeaderParam("Content-Type")`
+using the `contentType` from `SecretSetParameters` or `SecretUpdateParameters` as the value, but fixes the issue as the
+`@BodyParam` will set the content type to `application/json` as expected.
+
+```yaml
+directive:
+  - from: secrets.json
+    where: $.definitions
+    transform: >
+      $.SecretSetParameters.properties.contentType["x-ms-client-name"] = "secretContentType";
+      $.SecretUpdateParameters.properties.contentType["x-ms-client-name"] = "secretContentType";
```

Comment:
This is a bit unfortunate but at least this is a workaround. Could you please file an issue on the generator to not need this?

Reply:
Yeah, I'm not sure if this is an Autorest.java problem or somewhere else